docs/pages/management/infrastructure-as-code/terraform-starter/enroll-resources.mdx

```diff
@@ -0,0 +1,749 @@
+```
+
+### Virtual machine instances
+
+Each cloud-specific child module of `terraform-starter` declares resources to
+deploy a virtual machine instance on your cloud provider:
+
+<Tabs>
+<TabItem label="AWS">
+
+`ec2-instance.tf` declares a data source for an Amazon Linux 2023 machine image
+and uses it to launch EC2 instances that run Teleport Agents with the
+`teleport_provision_token` resource:
+
+```hcl
+(!examples/terraform-starter/aws/ec2-instance.tf!)
+```
+
+</TabItem>
+<TabItem label="Google Cloud">
+
+`gcp-instance.tf` declares Google Compute Engine instances that use the
+`teleport_provision_token` to run Teleport Agents:
+
+```hcl
+(!examples/terraform-starter/gcp/gcp-instance.tf!)
+```
+
+</TabItem>
+<TabItem label="Azure">
+
+`azure-instance.tf` declares an Azure virtual machine resource to run Teleport
+Agents using the `teleport_provision_token` resource, plus the required network
+interface for each instance. 
+
+Note that while Azure VM instances require a user account, this module declares
+a temporary one to pass validation, but uses Teleport to enable access to the
+instances:
+
+```hcl
+(!examples/terraform-starter/azure/azure-instance.tf!)
+```
+
+</TabItem>
+</Tabs>
+
+## Next steps: More options for enrolling resources
+
+In Part One of the Terraform starter guide, we showed you how to use Terraform
+to deploy a pool of Teleport Agents in order to enroll infrastructure resources
+with Teleport. While the guide showed you how to enroll resources dynamically,
+by declaring Terraform resources for each infrastructure resource you want to
+enroll, you can protect more of your infrastructure with Teleport by:
+
+- Configuring Auto-Discovery
+- Configuring static resource enrollment
+
+### Configure Auto-Discovery
+
+For a more scalable approach to enrolling resources than the one shown in this
+guide, configure the Teleport Discovery Service to automatically detect
+resources in your infrastructure and enroll them with the Teleport Auth Service.
+
+To configure the Teleport Discovery Service, edit the userdata script run by the
+Agent instances managed in the Terraform starter module. Follow the
+[Auto-Discovery
+guides](../../../enroll-resources/auto-discovery/auto-discovery.mdx) guides to
+configure the Discovery Service and enable your Agents to proxy the resources
+that the service enrolls.
+
+### Enroll static resources
+
+You can enroll resources statically by editing the configuration file that each
+Teleport Agent reads when it starts up, which is `/etc/teleport.yaml` by
+default. For information about enrolling static resources, read the
+documentation section for each kind of resource you would like to enroll:
+
+- [Databases](../../../enroll-resources/database-access/database-access.mdx)
+- [Windows
+  desktops](../../../enroll-resources/desktop-access/desktop-access.mdx)
+- [Kubernetes
+  clusters](../../../enroll-resources/kubernetes-access/kubernetes-access.mdx)
```

Comment:
Windows desktop and Kube clusters are not supported in the provider:
- windows dekstop, the resource is missing an abstraction (you cannot declare a windows desktop and leave the correct agent pick it up, you have to hard code the agent name)
- kube clusters, there is not technical reason, it's just not implemented because the feature was not priotitized